tweet_recommendations/other_methods/dbscan_based_method.py

```diff
@@ -0,0 +1,143 @@
+import multiprocessing as mp
+from collections import Counter
+from itertools import chain
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+import tqdm
+from scipy.spatial import cKDTree as KDTree, distance
+from sklearn.cluster import DBSCAN
+from sklearn.metrics.pairwise import cosine_distances
+
+from tweet_recommendations.other_methods.method import Method
+
+# "Finally, we use one as the number of minPoints and the average value of this vector as epsilon."
+# according to publication, this value should be fixed to 1.
+MIN_SAMPLES = 1
+
+
+class DBScanBasedEstimator(Method):
+    def __init__(self, embedding_method: Method, verbose: bool = False):
```

Comment:
Rozumiem, że embedding method to konkretnie SIFEmbedding? Jeśli tak to dlaczego anotacja typu na to nie wskazuje?

Reply:
A swoją drogą, jeśli to zawsze będzie SIF, to po co w ogóle dawać to jako parametr?

Reply:
Zaszłość. W przypadku, jak w zamyśle miało być inwariantne na typ embeddowania tweetów, zostało to. Teraz rzeczywiście można to zmienić, bo jedyną taką metodą jest SIF.